prediction_market_agent/agents/replicate_to_omen_agent/omen_replicate.py

```diff
@@ -132,23 +156,46 @@ def omen_replicate_from_tx(
         # Do as the last steps, because it calls OpenAI (costly & slow).
         if is_invalid(market.question):
             logger.info(
-                f"Skipping `{market.question}` because it seems to be an invalid question."
+                f"Skipping `{market.question}` was marked as invalid. Trying to rephrase and make it valid."
             )
-            continue
+            # We try rephrasing the question to make it valid, and run the validity check again.
+            new_question = rephrase(market.question)
+            logger.info(f"Rephrased `{market.question}` to `{new_question}`.")
+            if is_invalid(new_question):
+                logger.info(
+                    f"Skipping `{new_question}` because it could not be rephrased into a valid question."
+                )
+                continue
+            else:
+                market.question = new_question
+                was_rephrased = True
 
         if not is_predictable_binary(market.question):
             logger.info(
                 f"Skipping `{market.question}` because it seems to not be predictable."
             )
             continue
 
-        if market.description and not is_predictable_without_description(
-            market.question, market.description
+        # We don't check this for the was_rephrased case because the question was already considered predictable.
```

Comment:
After the rephrasing above, the question is not invalid, but it doesn't mean it's "predictable without description".

Seems to be this should be checked for even if it was rephrased.

Reply:
makes sense - now even after rephrasing, we still run `is_predictable_without_description`